pkg/apis/core/validation/shoot.go

```diff
@@ -556,6 +566,31 @@ func ValidateTotalNodeCountWithPodCIDR(shoot *core.Shoot) field.ErrorList {
 	return allErrs
 }
 
+// ValidateEncryptionConfigUpdate validates the updates to the KubeAPIServer encryption configuration.
+func ValidateEncryptionConfigUpdate(newConfig, oldConfig *core.KubeAPIServerConfig, etcdEncryptionKeyRotation *core.ETCDEncryptionKeyRotation, fldPath *field.Path) field.ErrorList {
```

Comment:
I guess we also want to forbid changes when the shoot is hibernating?
Also, when the spec!=status, we want to forbid hibernating the shoot, maybe? Or would the logic work to cater with proper encryption/decryption after the shoot gets woken up again?